src/components/VisualQueryEditor.tsx

```diff
@@ -175,10 +184,33 @@ export const VisualQueryEditor: React.FC<Props> = props => {
     [onChangeQuery, table, database, query]
   );
 
+  const onChangeTimeshift = useCallback(
+    (expression: QueryEditorExpression) => {
+      if (!isFieldExpression(expression) || !table) {
+        return;
+      }
+
+      const next = {
+        ...defaultQuery.expression,
+        ...query.expression,
+        from: table,
+        timeshift: expression,
+      };
+
+      onChangeQuery({
+        ...query,
+        resultFormat: resultFormat,
+        database: database,
+        expression: next,
+      });
+    },
+    [onChangeQuery, query, resultFormat, database, table]
```

Comment:
nit: Missing defaultQuery.expression and query.expression in dependencies.
```suggestion
    [onChangeQuery, query, resultFormat, database, table]
```

Reply:
hmm, the defaultQuery is immutable and should not be changed.

Reply:
Looks like react-hooks eslint recommend not to add them since changing them will not result in a re-render.